### ISSUE TITLE
Unable to deploy with NODE_ENV=test. Error: Cannot find module 'mongodb-memory-server'

### ISSUE THREAD

**Issue by vibhor-web-headout** [payloadcms/payload#7191]
### Describe the Bug

When deploying with NODE_ENV=test, the following is encountered:
```
Error: Cannot find module 'mongodb-memory-server'Require stack:
- /home/node/app/node_modules/@payloadcms/db-mongodb/dist/connect.js
- /home/node/app/node_modules/@payloadcms/db-mongodb/dist/index.js
- /home/node/app/dist/payload.config.js
- /home/node/app/node_modules/payload/dist/config/load.js
- /home/node/app/node_modules/payload/dist/payload.js
- /home/node/app/node_modules/payload/dist/initHTTP.js
- /home/node/app/node_modules/payload/dist/index.js
- /home/node/app/dist/server.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1145:15)
    at Module._load (node:internal/modules/cjs/loader:986:27)
    at Module.require (node:internal/modules/cjs/loader:1233:19)
    at Hook._require.Module.require (/home/node/app/node_modules/@newrelic/ritm/index.js:167:34)
    at require (node:internal/modules/helpers:179:18)
    at Object.connect (/home/node/app/node_modules/@payloadcms/db-mongodb/dist/connect.js:36:44)
    at BasePayload.init (/home/node/app/node_modules/payload/dist/payload.js:222:27)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async getPayload (/home/node/app/node_modules/payload/dist/payload.js:266:26)
    at async initHTTP (/home/node/app/node_modules/payload/dist/initHTTP.js:39:21)
{
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/node/app/node_modules/@payloadcms/db-mongodb/dist/connect.js',
    '/home/node/app/node_modules/@payloadcms/db-mongodb/dist/index.js',
    '/home/node/app/dist/payload.config.js',
    '/home/node/app/node_modules/payload/dist/config/load.js',
    '/home/node/app/node_modules/payload/dist/payload.js',
    '/home/node/app/node_modules/payload/dist/initHTTP.js',
    '/home/node/app/node_modules/payload/dist/index.js',
    '/home/node/app/dist/server.js'
  ]
}
Node.js v20.15.1
```

This seems to be coming from `@payloadcms/db-mongodb`. I couldn't find the env variable details anywhere.
I believe it expects an env variable `PAYLOAD_TEST_MONGO_URL` and if not, I guess it tries to create a MongoDB instance.

We are already passing the MongoDB URL in the payload config for the respective env.

### To Reproduce

Deploy payload with `NODE_ENV=test`

### Payload Version

2.22.0

### Adapters and Plugins

@payloadcms/db-mongodb: ^1.5.2

**Comment by denolfe** [comment]
Out of curiosity, what is the use case for deploying with `NODE_ENV` of test? 

Payload uses this variable to determine if a memory server should be used or not. 

You'll either need to set `PAYLOAD_TEST_MONGO_URL` or use a different `NODE_ENV`.

**Comment by vibhor-web-headout** [comment]
We will be hosting payload server for `production` & `test` environment for our internal tools.
This is something we encountered during `test` deployment.

> You'll either need to set PAYLOAD_TEST_MONGO_URL or use a different NODE_ENV.
Got it 👍 . We went with adding `PAYLOAD_TEST_MONGO_URL` to our env-variables.
I am assuming `test` as NODE_ENV will be common in repo's across. Possible to add this to the [docs](https://payloadcms.com/docs/getting-started/what-is-payload) as well ?
